components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -0,0 +1,99 @@
+package mozilla.components.service.glean.scheduler
+
+import android.app.job.JobInfo
+import android.app.job.JobParameters
+import android.app.job.JobScheduler
+import android.app.job.JobService
+import android.content.ComponentName
+import android.content.Context
+import kotlinx.coroutines.Job
+import mozilla.components.service.glean.Glean
+import java.util.concurrent.TimeUnit
+
+class MetricsPingScheduler : JobService() {
+    private var metricsPingJob: Job? = null
+
+    companion object {
+        // Job ID is an arbitrary identifier for our job once it's been scheduled.  It is the key
+        // that allows us to cancel or make changes to the job.  If you start a new job with the
+        // same ID, it will replace the existing job with the new one.  In order to prevent any
+        // future ping types from having the same job ID, the ordinal value of the ping event
+        // enum is used for unique identification.
+        val JOB_ID = Glean.PingEvent.Default.ordinal
```

Comment:
Note that job IDs are application scoped. This may need some kind of negotiating with the app - or use very predictive IDs. Otherwise we may end up overriding jobs from the app or the other way around.

This is one of the pitfalls `WorkManager` handles for you :)